src/bluesearch/entrypoint/database/topic_filter.py

```diff
@@ -0,0 +1,188 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+"""Filter articles with relevant topics."""
+from __future__ import annotations
+
+import argparse
+import logging
+import typing
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+
+from bluesearch.database.topic_info import TopicInfo
+from bluesearch.database.topic_rule import TopicRule, check_accepted
+
+logger = logging.getLogger(__name__)
+
+
+def init_parser(parser: argparse.ArgumentParser) -> argparse.ArgumentParser:
+    """Initialise the argument parser for the topic-filter subcommand.
+
+    Parameters
+    ----------
+    parser
+        The argument parser to initialise.
+
+    Returns
+    -------
+    argparse.ArgumentParser
+        The initialised argument parser. The same object as the `parser`
+        argument.
+    """
+    parser.description = "Filter articles with relevant topics"
+
+    parser.add_argument(
+        "extracted_topics",
+        type=Path,
+        help="""
+        Path to a .JSONL file that was an output of the `topic-extract`
+        command.
+        """,
+    )
+    parser.add_argument(
+        "filter_config",
+        type=Path,
+        help="""
+        Path to a .JSONL file that defines all the rules for filtering.
+        """,
+    )
+    parser.add_argument(
+        "output_file",
+        type=Path,
+        help="""
+        Path to a .CSV file where rows are different articles
+        and columns contain relevant information about these articles.
+        """,
+    )
+
+    return parser
+
+
+def parse_filter_config(config: list[dict]) -> tuple[list[TopicRule], list[TopicRule]]:
+    """Parse filter configuration.
+
+    Parameters
+    ----------
+    config
+        Topic Rules configuration
+
+    Returns
+    -------
+    topic_rules_accept : list[TopicRule]
+        List of accepted TopicRule
+    topic_rules_reject : list[TopicRule]
+        List of rejected TopicRule
+
+    Raises
+    ------
+    ValueError
+        If one of the label value is different from accept and reject.
+    """
+    topic_rules_accept, topic_rules_reject = [], []
+    for raw_rule in config:
+        rule = TopicRule(
+            level=raw_rule.get("level"),
+            source=raw_rule.get("source"),
+            pattern=raw_rule.get("pattern"),
+        )
+        label = raw_rule["label"]
+
+        if label == "accept":
+            topic_rules_accept.append(rule)
+        elif label == "reject":
+            topic_rules_reject.append(rule)
+        else:
+            raise ValueError(f"Unsupported label {label}")
+
+    return topic_rules_accept, topic_rules_reject
+
+
+def filter_topics(
```

Comment:
Personally, I think that it would help with clarity if  the various `blueasearch/entrypoint/database/*.py` only define the entrypoints interface and contain an `init_parser` and a `run()`, while the main logic and functionalities are implemented in `blueasearch/database/*.py`.

So maybe we can move `filter_topics()` and `parse_filter_configs()` to `blueasearch/database/topic_rule.py`?
An exception could be made if you think that those two functions are simple utils that only make logical sense in the context of this entrypoint (but at least for `filter_topics()` I feel that's not really the case), in which case I think it's OK to keep them here, maybe just make them private/internal with a leading `_`  underscore.

Reply:
I don't really have a strong opinion on this. What about @EmilieDel @Stannislav ?

Reply:
It's not a huge deal, and I don't want to block this PR for this detail. So feel free to ignore this thread and move on! :)

Just for the future, I believe it's usually cleaner to separate the **_interface_** (CLI, widget, GUI, ...) from the **_implementation_** of the functionality (functions, classes, ...) by putting them into separate files/modules.

Reply:
Sure, let's move it to `bluesearch.database` :)